### ISSUE TITLE
V2 - ais-refinement-list component: "TypeError: this.client.initIndex is not a function" When trying to use the searchbox

### ISSUE THREAD

**Issue by tfossoul** [algolia/vue-instantsearch#545]
**Feedback on the v2 alpha version 🆕**

I'm just trying to search something on my refinement-list and I got an error on the console.

Here my definition:
```
<ais-refinement-list attribute="Country.n" operator="and" searchable showMore></ais-refinement-list>
```
Here is where the error appear:
```
var searchForFacetValuesPromise = typeof this.client.searchForFacetValues === 'function'
    ? this.client.searchForFacetValues([{indexName: state.index, params: algoliaQuery}])
    : this.client.initIndex(state.index).searchForFacetValues(algoliaQuery);
```

Any idea ?

By the way on your documentation you explain that we should specify the attribute in this way ":attribute="brands" but it doesn't work. "attribute="brands"" works.

**Comment by Haroenv** [comment]
Hmm, that seems to be a bug in the documentation, if it's a string, you don't need to use v-bind, since then you'd evaluate it.

Can you make a sandbox that errors like that though? I'd expect the prop type to fail before something else :)

**Comment by tfossoul** [comment]
Everythings is working fine when I'm not trying to use our own backend and do direct request.

**Comment by Haroenv** [comment]
I see what's going on. You need to add `searchForFacetValues() { return /* a promise */ }` in your custom client instantiation. This is used for the `searchable` prop on the refinementList. 

Please see the guide here: https://community.algolia.com/instantsearch.js/v2/guides/custom-backend.html#supporting-search-for-facet-values

**Comment by tfossoul** [comment]
@Haroenv it is working. But the search is send directly to algolia instead of passing through my backend..

**Comment by Haroenv** [comment]
Can you give the code you are using please? I think something weird is happening there. Note that I gave pseudo code, see the guide I linked on how to use it with your backend.

**Comment by tfossoul** [comment]
I just try these 2 options. I put a breakpoint and it appear these are never used..
http://localhost:4000/search is my local backend...

 `searchForFacetValues () { return /* a promise */ }`
`
  searchForFacetValues (requests) {
    return fetch('http://localhost:4000/search', {
      method: 'post',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ requests })
    }).then(res => res.json())
  }
`

**Comment by Haroenv** [comment]
Can you **please** give a **full** example? I can't work with these half snippets

**Comment by tfossoul** [comment]
Correction. I did an error on my code. So searchForFacetValues is called.

Now I got the following error on my backend:

(node:5828) UnhandledPromiseRejectionWarning: AlgoliaSearchError: Unknown parame
ter: facetName
    at success (....proxy\node_modules\algoliasearch\s
rc\AlgoliaSearchCore.js:375:32)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
(node:5828) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This
error originated either by throwing inside of an async function without a catch
block, or by rejecting a promise which was not handled with .catch(). (rejection
 id: 28)

Use the package "algoliasearch": "^3.30.0".
